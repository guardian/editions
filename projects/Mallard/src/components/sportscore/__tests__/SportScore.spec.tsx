import React from 'react';
import type { ReactTestRendererJSON } from 'react-test-renderer';
import TestRenderer from 'react-test-renderer';
import { SportScore } from 'src/components/SportScore/SportScore';

const sportScore = 'Luton 3 - 0 Watford';

describe('SportScore', () => {
	it('should show a SportScore with default styling', () => {
		const component: ReactTestRendererJSON | null = TestRenderer.create(
			<SportScore sportScore={sportScore} />,
		).toJSON();
		expect(component).toMatchSnapshot();
	});

	it('should show a SportScore with trailImage styling', () => {
		const component: ReactTestRendererJSON | null = TestRenderer.create(
			<SportScore sportScore={sportScore} type="trailImage" />,
		).toJSON();
		expect(component).toMatchSnapshot();
	});

	it('should show a SportScore with smallItems styling', () => {
		const component: ReactTestRendererJSON | null = TestRenderer.create(
			<SportScore sportScore={sportScore} type="smallItems" />,
		).toJSON();
		expect(component).toMatchSnapshot();
	});
});
