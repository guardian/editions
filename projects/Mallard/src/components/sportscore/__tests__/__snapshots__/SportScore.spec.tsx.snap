// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SportScore should show a SportScore with default styling 1`] = `
<View
  accessibilityLabel="Luton 3 - 0 Watford"
  style={
    Array [
      Object {
        "backgroundColor": "#ffe500",
        "padding": 0,
        "paddingBottom": 2,
        "paddingHorizontal": 4.666666666666667,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Object {
        "fontFamily": "GHGuardianHeadline-Regular",
        "fontSize": 16,
        "letterSpacing": 0.25,
        "lineHeight": 17,
      }
    }
  >
    Luton 3 - 0 Watford
  </Text>
</View>
`;

exports[`SportScore should show a SportScore with smallItems styling 1`] = `
<View
  accessibilityLabel="Luton 3 - 0 Watford"
  style={
    Array [
      Object {
        "backgroundColor": "#ffe500",
        "padding": 0,
        "paddingBottom": 2,
        "paddingHorizontal": 4.666666666666667,
      },
      Object {
        "flex": 0,
      },
    ]
  }
>
  <Text
    style={
      Object {
        "fontFamily": "GHGuardianHeadline-Regular",
        "fontSize": 16,
        "letterSpacing": 0.25,
        "lineHeight": 17,
      }
    }
  >
    Luton 3 - 0 Watford
  </Text>
</View>
`;

exports[`SportScore should show a SportScore with trailImage styling 1`] = `
<View
  accessibilityLabel="Luton 3 - 0 Watford"
  style={
    Array [
      Object {
        "backgroundColor": "#ffe500",
        "padding": 0,
        "paddingBottom": 2,
        "paddingHorizontal": 4.666666666666667,
      },
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
      },
    ]
  }
>
  <Text
    style={
      Object {
        "fontFamily": "GHGuardianHeadline-Regular",
        "fontSize": 16,
        "letterSpacing": 0.25,
        "lineHeight": 17,
      }
    }
  >
    Luton 3 - 0 Watford
  </Text>
</View>
`;
